src/components/Premium/Premium.tsx

```diff
@@ -4,44 +4,92 @@ import Payment from "../Payment/Payment";
+            }
+            setHasPremiumUntil(hasPremiumUntil)
+        });
+    }
+
 
     function onLogin() {
         let googleId = localStorage.getItem('googleId');
         if (googleId) {
             setIsLoggedIn(true);
+            loadHasPremiumUntil();
         }
     }
 
+    function closeFeatureDialog() {
+        setShowFeatureDialog(false);
+    }
+
+    let featureDialog = (
+        <Modal show={showFeatureDialog} onHide={closeFeatureDialog}>
+            <Modal.Header closeButton>
+                <Modal.Title>Features</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+                <PremiumFeatures />
+            </Modal.Body>
+        </Modal>
+    );
+
     return (
         <div className="premium">
-            <h4>
+            <h2>
                 <NavBar />
                 Premium
-            </h4>
+            </h2>
+            <hr />
+            {hasPremium ? <p style={{ color: "#00bc8c" }}>You have a premium account. Thank you for your support.</p> : <p style={{ color: "red" }}>You do no have a premium account</p>}
+            {
+                hasPremium ?
+                    <div>
+                        <OverlayTrigger
+                            overlay={<Tooltip id={generateUUID()}>
+                                <span>{hasPremiumUntil?.toDateString()}</span>
+                            </Tooltip>}>
+                            <p>Your premium ends: {moment(hasPremiumUntil).fromNow()}</p>
+                        </OverlayTrigger>
+                    </div> : ""
+            }
+            <hr />
             <Card className="premium-card">
-                <p>Premium users get access to the following features:</p>
-                <ul>
-                    <li>Up to 100 subscriptions</li>
-                    <li>Full access to the flipper</li>
-                    <li>Get good flipps faster</li>
-                    <li>Our "thank you"</li>
-                    <li>Priority feature request</li>
-                </ul>
+                {hasPremium
+                    ? <p>Thank you for your support. You have a Premium account. By buying another Premium-Plan you can extend your premium-time.
+                    You can use the following premium-features:
+                    </p>
+                    : <p>Buy Premium to support us and get access to these exclusive features:</p>}
+                <span style={{ cursor: "pointer", width: "fit-content", fontWeight: "bold" }} onClick={() => { setShowFeatureDialog(true) }}><LinkIcon /> Show features</span>
             </Card>
             <div>
                 <GoogleSignIn onAfterLogin={onLogin} />
-                {isLoggedIn ? <Payment /> : ""}
+                {isLoggedIn ? <Payment hasPremium={hasPremium} /> : ""}
                 {wasAlreadyLoggedInGoogle && !isLoggedIn ? getLoadingElement() : ""}
                 {!wasAlreadyLoggedInGoogle && !isLoggedIn ? <p>To use subscriptions please login with Google:</p> : ""}
```

Comment:
This text should be above the `GoogleSignIn` Element and state
To use premium please login with Google